src/ripple/app/misc/impl/ValidatorList.cpp

```diff
@@ -17,174 +17,744 @@
 */
 //==============================================================================
 
-#include <BeastConfig.h>
+#include <ripple/app/misc/detail/WorkPlain.h>
+#include <ripple/app/misc/detail/WorkSSL.h>
 #include <ripple/app/misc/ValidatorList.h>
 #include <ripple/basics/Slice.h>
-#include <ripple/basics/StringUtilities.h>
+#include <ripple/beast/rfc2616.h>
+#include <ripple/json/json_reader.h>
+#include <beast/core/detail/base64.hpp>
+#include <beast/core/placeholders.hpp>
 #include <boost/regex.hpp>
 
 namespace ripple {
 
-ValidatorList::ValidatorList (beast::Journal j)
-    : j_ (j)
+// default site query frequency - 5 minutes
+auto constexpr DEFAULT_REFRESH_INTERVAL = std::chrono::minutes{5};
+
+ValidatorList::ValidatorList (
+    ManifestCache& manifests,
+    boost::asio::io_service& io_service,
+    beast::Journal j)
+    : manifests_ (manifests)
+    , j_ (j)
+    , io_service_ (io_service)
+    , timer_ (io_service)
+    , fetching_ (false)
+    , pending_ (false)
+    , stopping_ (false)
+    , fetchedLists_ (0)
+    , quorum_ (0)
 {
 }
 
-boost::optional<std::string>
-ValidatorList::member (PublicKey const& identity) const
+ValidatorList::~ValidatorList()
 {
-    std::lock_guard <std::mutex> sl (mutex_);
+    if (! stopping_)
+        stop();
+    std::unique_lock<std::mutex> lock{mutex_};
+    cv_.wait(lock, [&]{ return ! fetching_; });
+}
+
+bool
+ValidatorList::load (
+    PublicKey const& localSigningKey,
+    std::vector<std::string> const& configKeys,
+    std::vector<std::string> const& siteURIs,
+    std::vector<std::string> const& publisherKeys,
+    std::vector<std::string> const& configManifest)
+{
+    static boost::regex const re (
```

Comment:
I have a memory of us trying to use `std::regex` a year or two ago and having to revert back to `boost` because of inconsistencies.  I don't recall if that was this use or not though.  Generally I like to prefer `std` over `boost` when given a choice.  But it would take some careful testing to convert `regex` due to its known difficulty of implementation and known inconsistency among std::implementations.   We probably need better motivation for such an effort than "prefer `std` to `boost`.

I would have no objection if somebody wants to take that task on.  But I don't see it as a high priority or big pay off.